examples/hello_world/client/nodejs/index.js

```diff
@@ -1,24 +1,54 @@
 const fs = require('fs');
-const protoLoader = require('@grpc/proto-loader');
+const protobufjs = require('protobufjs');
 const grpc = require('@grpc/grpc-js');
+const grpcProtoLoader = require('@grpc/proto-loader');
 
 const CERT_PATH = __dirname + '/../../../certs/local/ca.pem';
-const PROTO_PAH = __dirname + '/../../proto/hello_world.proto';
-
-const packageDefinition = protoLoader.loadSync(PROTO_PAH);
-const helloWorldProto = grpc.loadPackageDefinition(packageDefinition).oak
-  .examples.hello_world;
-const credentials = grpc.credentials.createSsl(fs.readFileSync(CERT_PATH));
-
-// TODO(#1066): Use a more restrictive Label.
-const client = new helloWorldProto.HelloWorld('localhost:8080', credentials);
-
-client.sayHello({ greeting: 'Node.js' }, (error, response) => {
-  if (error) {
-    console.error(error);
-    process.exit(1);
-  } else {
-    console.log(response.reply);
-    process.exit(0);
+const SERVICE_PROTO_PAH = __dirname + '/../../proto/hello_world.proto';
+const OAK_LABEL_PROTO_PAH = __dirname + '/../../../../oak/proto/label.proto';
+
+// Keep in sync with /oak/server/rust/oak_runtime/src/node/grpc/server/mod.rs.
+const oakLabelGrpcMetadataKey = 'x-oak-label-bin';
+
+async function main() {
+  const [oaklabelDefinition, helloWorldDefinition] = await Promise.all([
+    protobufjs.load(OAK_LABEL_PROTO_PAH),
+    grpcProtoLoader.load(SERVICE_PROTO_PAH),
+  ]);
+
+  function getGrpcMetadata() {
+    // TODO(#1097): move this into an SDK package to allow re-use.
+    const OakLabel = oaklabelDefinition.lookupType('oak.label.Label');
+
+    // TODO(#1066): Use a more restrictive Label.
+    const publicUntrustedLabel = OakLabel.create({});
+    const encodedLabel = OakLabel.encode(publicUntrustedLabel).finish();
+
+    const metaData = new grpc.Metadata();
+    metaData.add(oakLabelGrpcMetadataKey, encodedLabel);
+
+    return metaData;
   }
-});
+
+  const helloWorldProto = grpc.loadPackageDefinition(helloWorldDefinition).oak
+    .examples.hello_world;
+  const credentials = grpc.credentials.createSsl(fs.readFileSync(CERT_PATH));
+
+  const client = new helloWorldProto.HelloWorld('localhost:8080', credentials);
+
+  client.sayHello(
+    { greeting: 'Node.js' },
+    getGrpcMetadata(),
```

Comment:
Could you please add comments what are these arguments?
Because currently it may be a little bit confusing what exactly we are passing to this function.

Reply:
Can do. :) 

It's mostly just using the nodejs grpc API, but can link to it & add comments

Reply:
Added. :) Please do let me know if they are helpful.